plugin/discovery/version_set.go

```diff
@@ -9,52 +9,49 @@ import (
 // obtain a real Constraint object, or discover that it is invalid.
 type ConstraintStr string
 
-// Parse transforms a ConstraintStr into a VersionSet if it is
+// Parse transforms a ConstraintStr into a Constraints if it is
 // syntactically valid. If it isn't then an error is returned instead.
-func (s ConstraintStr) Parse() (VersionSet, error) {
+func (s ConstraintStr) Parse() (Constraints, error) {
 	raw, err := version.NewConstraint(string(s))
 	if err != nil {
-		return VersionSet{}, err
+		return Constraints{}, err
 	}
-	return VersionSet{raw}, nil
+	return Constraints{raw}, nil
 }
 
 // MustParse is like Parse but it panics if the constraint string is invalid.
-func (s ConstraintStr) MustParse() VersionSet {
+func (s ConstraintStr) MustParse() Constraints {
 	ret, err := s.Parse()
 	if err != nil {
 		panic(err)
 	}
 	return ret
 }
 
-// VersionSet represents a set of versions which any given Version is either
+// Constraints represents a set of versions which any given Version is either
 // a member of or not.
-type VersionSet struct {
-	// Internally a version set is actually a list of constraints that
-	// *remove* versions from the set. Thus a VersionSet with an empty
-	// Constraints list would be one that contains *all* versions.
+type Constraints struct {
 	raw version.Constraints
 }
 
-// AllVersions is a VersionSet containing all versions
-var AllVersions VersionSet
+// AllVersions is a Constraints containing all versions
+var AllVersions Constraints
```

Comment:
I think AllVersions works the same as NoConstraints in this case. I do agree with your naming comments below though. I just did a GoRename of the type and didn't go back to check the methods.